File: 프로그래머스/lv2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    result = []
    id_name={}
    
    for i in record:
        if "Leave" not in i:          
            string=i.split()            
            id_name[string[1]]=string[2]
             
    for i in record:
        name=i.split()
        if "Enter" in i:
            result.append("%s님이 들어왔습니다."%id_name[name[1]])
        elif "Leave" in i:
            result.append("%s님이 나갔습니다."%id_name[name[1]])
        else:
            pass
         
    return result

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
// record 돌면서 Enter거나 change면 map에 저장, Leave면 건너뛰기, 
// 다시 돌면서 id에 해당하는 닉네임 출력

function solution(record) {
    const maps = {}
    const result = [];
    
    for (i of record){
        const [type,id,name] = i.split(' ')
        if (type == 'Enter' || type == 'Change'){
            maps[id] = name
        }
    }
    for (i of record){
        const [type,id,name] = i.split(' ')
        if (type == 'Enter'){
            result.push(`${maps[id]}님이 들어왔습니다.`)
        } else if (type == 'Leave'){
            result.push(`${maps[id]}님이 나갔습니다.`)
        }
    }
    return result
}